searchmenu/init.lua

```diff
@@ -14,15 +14,15 @@ local add_binds, add_cmds = modes.add_binds, modes.add_cmds
 local menu_binds = binds.menu_binds
 local new_mode = require("modes").new_mode
 
-local function try_search_engine (url, selection)
-    local ret
-    if type(url) == "string" then
-        ret = string.format(url, luakit.uri_encode(selection))
-    elseif pcall(url, selection) then
-        ret = url(selection)
+local function try_search_engine (engine, selection)
+    if type(engine) == "string" then
+        return string.format(engine, luakit.uri_encode(selection))
+    else
+      local _, ret = pcall(engine, selection)
+      return ret
```

Comment:
This should be:
```lua
local ok, ret = pcall(...)
return ok and ret or nil
```

Reply:
If `pcall` fails, `ret` is `nil` anyway, so what's the difference?

Reply:
It'll be an error message, not nil

Reply:
Ah yes, stupid me !